.github/workflows/scripts/slack_gem_notifications/notifications_methods.rb

```diff
@@ -39,10 +39,22 @@ def gem_updated?(versions)
   Time.now.utc - Time.parse(versions[0]['created_at']) < 24 * 60 * 60
 end
 
+def github_diff(gem_name, newest, previous)
+  diff = HTTParty.get("https://github.com/#{gem_name}/#{gem_name}/compare/v#{previous}...v#{newest}")
+
+  diff.success?
+end
+
 def send_bot(gem_name, versions)
-  abort "Expected exactly 2 version numbers in the 'versions' array" unless versions.size == 2
+  abort("Expected exactly 2 version numbers in the 'versions' array") unless versions.size == 2
   newest, previous = versions[0]['number'], versions[1]['number']
-  HTTParty.post(ENV['SLACK_GEM_NOTIFICATIONS_WEBHOOK'],
-    headers: {'Content-Type' => 'application/json'},
-    body: {text: "A new gem version is out! Gem: #{gem_name}, #{previous} -> #{newest}."}.to_json)
+  if github_diff(gem_name, newest, previous)
+    HTTParty.post(ENV['SLACK_GEM_NOTIFICATIONS_WEBHOOK'],
+      headers: {'Content-Type' => 'application/json'},
+      body: {text: "A new gem version is out :sparkles: <https://rubygems.org/gems/#{gem_name}|*#{gem_name}*>, #{previous} -> #{newest}\n\n<https://github.com/#{gem_name}/#{gem_name}/compare/v#{previous}...v#{newest}|Check out the git diff>"}.to_json)
+  else
```

Comment:
There's a good amount of content shared between the `if` and `else` blocks. From what I can tell, it's really just the `text` value being set in the JSON body that changes. Maybe we could build the text string conditionally with `if/else` but then share all of the common code by only having it appear once (DRY).

Reply:
addressed with [f9d253d](https://github.com/newrelic/newrelic-ruby-agent/pull/1273/commits/f9d253def2866a284602c817d379bd83621c02cc)